### ISSUE TITLE
Problematic observations from recent engineering time, 20220413, 20220414, 20220416

### ISSUE THREAD

**Issue by schlafly** [desihub/desisurveyops#44]
This was discussed on Slack and by email, but I am unclear as to the status.  The beginning of the night on 20220413 featured engineering observations on main survey tiles.  We need to mark some of these exposures as bad.

QA for observations 20619/130091 and 25297/130094 look fine to me, though the night log reports that 25297/130094 did not have a turbulence correction applied.  I think we should keep these two observations.

QA for 22170/130095 looks terrible and this exposure needs to be marked bad and the tile reprocessed.  This duplicates https://github.com/desihub/desisurveyops/issues/43 and https://github.com/desihub/desisurveyops/issues/40 .

Night log reports observations of a few other tiles that may have already been marked bad and do not make the nightly QA (e.g., 22202, 22211, 22203, 20638).  Because there is no QA, I have not inspected these tiles.

On 20220414 the observers started observing backup tiles to avoid problems like the above.  But then they moved to BRIGHT tile 23075 and did more test observations.  It looks like we got 4 back to back short exposures there.
https://data.desi.lbl.gov/desi/spectro/redux/daily/tiles/cumulative/20375/20220414/tile-qa-20375-thru20220414.png
QA looks fine to me and I don't see major positioning problems like those we saw for other fast sequencer test observations.  It's certainly not normal survey observations to have 4 short exposures on a BRIGHT tile, though, so we could consider marking all of these as bad.  The rest of 20220414 looks fine to me.

On 20220417, I don't see anything dubious in the night log.  There's a weird exposure of 42426 in the nightqa.  It's weird because:
- the QA only shows one observation on 20220416, so why am I seeing it on 20220417?  (note: tile was also observed on 20220417, so maybe the observation on 20220417 was marked bad, but I'm not sure I want to see it in that case?)
- Positioning is bad on all petals except P0, so we'd want to mark the exposure as bad?
It looks to me like the 20220416 exposure on this tile was good, but the 20220417 exposure was bad.  But the QA script includes the flagged bad exposure on 20220417, making the positioning on all petals look bad despite the fact that this exposure was marked bad?  @araichoor .  
https://data.desi.lbl.gov/desi/spectro/redux/daily/tiles/cumulative/42426/20220416/tile-qa-42426-thru20220416.png
https://data.desi.lbl.gov/desi/spectro/redux/daily/tiles/cumulative/42426/20220417/tile-qa-42426-thru20220417.png
I don't know why we would have failed to position on all petals; that never happens normally?

Nothing else exceptionally alarming.

Short version: let's mark 22170/130095 bad and reprocess.  Let's discuss 23075, 20619, 25297, but probably keep all of these; their QA is fine.  Let's check to see if there's a bug in the QA script where data from bad, excluded exposures are included in the positioner offset plots, as suggested by 42426.

**Comment by araichoor** [comment]
about 42426 on 20220416/20220417:

I ve looked a bit, I m not 100% sure of what s going on.
but I suspect the issue comes from the fact that the daily cron job has not been run around those dates (see https://desisurvey.slack.com/archives/C01HNN87Y7J/p1650217810827989).

first, for instance, looking at the spectra-*log files, I see that the 20220417 redux does not use the 20220417 exposure, only the 20220416 one:
```
raichoor@cori03:~/Scripts> grep RUNNING /global/cfs/cdirs/desi/spectro/redux/daily/tiles/cumulative/42426/20220417/logs/spectra-0-42426-thru20220417.log
RUNNING srun -N 1 -n 1 -c 64 --cpu-bind=none desi_group_spectra --inframes exposures/20220416/00130563/cframe-b0-00130563.fits exposures/20220416/00130563/cframe-r0-00130563.fits exposures/20220416/00130563/cframe-z0-00130563.fits --outfile tiles/cumulative/42426/20220417/spectra-0-42426-thru20220417.fits --header SPGRP=cumulative SPGRPVAL=20220417 NIGHT=20220417 TILEID=42426 SPECTRO=0 PETAL=0
```
that explains why tile-qa-42426-thru20220417.png only reports the 20220416 exposure (it grabs the exposure list from the spectra*fits files).

then, about the disagreement between the two qa plots (20220416 vs. 20220417):
as those are both only using the 20220416 exposure, I d have expected those to be similar.
but the MEAN_DELTA_{XY} columns in the tile-qa-42426-thru2022041{6,7}.fits files are different, except for PETAL_LOC=0.
e.g.:
```
d16 = Table.read("/global/cfs/cdirs/desi/spectro/redux/daily/tiles/cumulative/42426/20220416/tile-qa-42426-thru20220416.fits","FIBERQA")
d17 = Table.read("/global/cfs/cdirs/desi/spectro/redux/daily/tiles/cumulative/42426/20220417/tile-qa-42426-thru20220417.fits","FIBERQA")
for petal in range(10):
    sel = d16["PETAL_LOC"] == petal
    print(petal, (d16["MEAN_DELTA_X"][sel] != d17["MEAN_DELTA_X"][sel]).sum())
=>
0 0
1 500
2 499
3 496
4 496
5 498
6 498
7 495
8 498
9 500
```
I don t understand how that could be.
I thought about a possible re-processing of the 20220416 exposure in between, but from the timestamps, it looks like not.
maybe @sbailey or @akremin could figure that out?

**Comment by djschlegel** [comment]
Just noting that some of the exposures from the night 20220413 were indeed already rejected as you've speculated.  That list here:
https://desi.lbl.gov/trac/tracmailman/browser/private/desi-survey/2022-April/003733.html

**Comment by araichoor** [comment]
thanks; though the exposure in question here is from 20220417.
if I m correct, rejected exposures have EFFTIME=0, and this exposure from 20220417 has `EFFTIME~4s`, so it does not look to be blacklisted.
*but*, I realize now that this low EFFTIME may be the reason why it got filtered out by the pipeline, as I think I recall there is a cut on low EFFTIME (btw, is the cut value lower for BACKUP tiles?).
that at least explains the first part of the puzzle.
```
d = Table.read("/global/cfs/cdirs/desi/spectro/redux/daily/exposures-daily.fits","EXPOSURES")
d[d["TILEID"]==42426]["NIGHT","EXPID","TILEID","EXPTIME","EFFTIME_SPEC"]
<Table length=2>
 NIGHT   EXPID  TILEID      EXPTIME          EFFTIME_SPEC  
 int32   int32  int32       float64            float64     
-------- ------ ------ ------------------ -----------------
20220416 130563  42426  600.0385131835938 72.46405029296875
20220417 130673  42426 60.013885498046875 4.093623638153076
```

**Comment by schlafly** [comment]
And, sorry to make this issue so multi-layered, but we need to decide whether or not to keep these tiles:
23075 - 4 back to back short exposures maybe with new sequencer on; no other reasons I know of to be unhappy with this tile
20619 - maybe observed with new sequencer
25297 - observed with new sequencer, night log says turbulence corrections are missing
Since these would all pass our normal QA metrics, I am inclined to keep them all, in which case we could do an MTL update tomorrow.  If we want to reflect more on these, we need to mark them as unsure to enable MTL updates in the interim.

**Comment by schlafly** [comment]
We accepted 23075, 20619, 25297 when we did MTL updates yesterday.  I'm fine with that.  Outstanding issue in this ticket is to resolve the inconsistent MEAN_DELTA_X, MEAN_DELTA_Y on 42426 where an excluded exposure is being included in the mean X and Y?

**Comment by araichoor** [comment]
about tileid=42426: I found the issue.
forget my previous comments, apologies.

the reason is that, the exposures list is inferred from looking at the first `spectra*fits` file found.
here, it is for petal 0, but that petal is discarded for the 20220417 exposure (130673); so the tile_qa/tile_qa_plot codes miss that exposure, which creates the apparent mismatch.

the plotted `MEAN_DELTA_X` and `MEAN_DELTA_Y` columns come from the redrock FIBERMAP extensions, so those consider both exposures for all petals, except for petal=0.

I m opening a PR with a fix...

thanks @schlafly for reminding me about that!

**Comment by araichoor** [comment]
PR here: https://github.com/desihub/desispec/pull/1767

for this tile, maybe we d want to blacklist expid= 130673, and rerun the daily pipeline?

**Comment by schlafly** [comment]
I think we just need to reprocess 20220417/130673, tileid 42426 to close this ticket.  @akremin, maybe?

**Comment by akremin** [comment]
Exp 130673 was marked to ignore, and the files removed at some point in the past. The claim was bad positioning, but I may have just been following Anand's May 9th message.

The `tiles-daily.csv` hadn't been updated, which led to the continued inconsistency. I have rerun that and the nightly QA. `tiles-daily.csv` now reflects the fact that 220416 is the `LASTNIGHT` and `exposures-daily.csv` reflects the 0s `EFFTIME_SPEC` for 20220417.

The one exposure on 20220416 got 72.5s `EFFTIME_SPEC`, so the tile remains completed even without these 4s of time.
